doc/src/guides/solving/solve-numerically.md

```diff
@@ -0,0 +1,216 @@
+>>> from sympy.abc import x
+>>> nsolve(x**2 - 1, -0.1)
+1.00000000000000
+```
+
+You can ensure the root found is in a given interval, if such a root exists,
+using `solver='bisect'` by specifying the interval in a tuple. Here, specifying
+the interval `(-10, 0)` ensures that the root `-1` is found:
+
+```py
+>>> from sympy import nsolve
+>>> from sympy.abc import x
+>>> nsolve(x**2 - 1, (-10, 0), solver='bisect')
+-1.00000000000000
+```
+
+## Solve a System of Equations Numerically
+
+### Solve Multidimensional Functions
+
+To solve multidimensional functions, supply a tuple of
+- functions `(f1, f2)`
+- variables to solve for `(x1, x2)`
+- starting values `(-1, 1)`
+
+```py
+>>> from sympy import Symbol, nsolve
+>>> x1 = Symbol('x1')
+>>> x2 = Symbol('x2')
+>>> f1 = 3 * x1**2 - 2 * x2**2 - 1
+>>> f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+>>> print(nsolve((f1, f2), (x1, x2), (-1, 1)))
+Matrix([[-1.19287309935246], [1.27844411169911]])
+```
+
+### Increase Precision of the Solution
+
+You can increase the precision of the solution using `prec`:
+
+```py
+>>> from sympy import Symbol, nsolve
+>>> x1 = Symbol('x1')
+>>> x2 = Symbol('x2')
+>>> f1 = 3 * x1**2 - 2 * x2**2 - 1
+>>> f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+>>> print(nsolve((f1, f2), (x1, x2), (-1, 1), prec=25))
+Matrix([[-1.192873099352460791205211], [1.278444111699106966687122]])
+```
+
+### Create a Function That Can Be Solved With SciPy
+
+As noted above, SymPy focuses on symbolic computation and is not optimized for
+numerical calculations. A recommend use case for SymPy in a workflow is as given
+below:
+- use SymPy to generate (by symbolically simplifying or solving an equation) the
+  mathematical expression
+- use a numerical library such as SciPy to generate numerical solutions
+
+```py
+>>> from sympy import simplify, cos, sin
+>>> from sympy.abc import x, y
+>>> expr = cos(x * (x + x**2)/(x*sin(y)**2 + x*cos(y)**2 + x))
+>>> simplify(expr)
+cos(x*(x + 1)/2)
+>>> from sympy.utilities.lambdify import implemented_function
+>>> from sympy import lambdify
+>>> f = implemented_function('f', lambda x: cos(x*(x + 1)/2))
+>>> lam_f = lambdify(x, f(x))
+>>> from scipy import optimize
+>>> sol = optimize.root_scalar(lam_f, bracket=[0, 2], method='brentq')
+>>> sol.root
+1.3416277185114782
+```
+
+## Use the Solution Result
+
+### Substitute the Result Into an Expression
+
+The best practice is to use {func}`~sympy.core.evalf` to to substitute numerical
+values into expressions. Using [`subs`](sympy.core.basic.Basic.subs) can give
+incorrect result due to precision errors:
+
+```py
+>>> from sympy import cos, nsolve, Symbol
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> y = Symbol('y')
+>>> z = Symbol('z')
+>>> values = {x: x_value, y: 1e16, z: 1e16}
+>>> (x + y - z).evalf(subs=values) # Gives correct result
+0.739085133215161
+>>> (x + y - z).subs(values) # Gives incorrect result due to precision errors
```

Comment:
This example (and the one below) seem a bit odd because you are substituting the solution from an equation into another expression (`x + y - z`) and it's not clear why you would do that. In fact, the issue here has nothing to do with the value 0.73... but rather from the cancellation from 1e16 and -1e16.

Maybe a more motivated example would be showing substituting the solved value back into the equation. This has the same fundamental issue because of the subtraction in `cos(x) - x`.

```py
>>> f.subs(x, x_value)
0
>>> f.evalf(subs={x: x_value})
-5.12757857962640e-17
```

Here the second answer is more correct, because the `x_value` is not an exact solution to the equation, so substituting it should not give exactly 0.

Reply:
That [example was suggested](https://github.com/sympy/sympy/pull/23853#discussion_r933978731) by @smichr from the [EvalfMixin](https://docs.sympy.org/latest/modules/core.html?highlight=subs#sympy.core.evalf.EvalfMixin) docstring, which notes "Using the subs argument for evalf is the accurate way to evaluate such an expression."

It doesn't strike me as odd to substitute the solution from an equation into another expressions--it seems like that's something you'd do when you have one equation that you can use to solve for `x`, and then want to use that value of `x` to find the value of another expression.

If desired, I could return to the docstring's formulation where the value (that we don't want to get wiped out by a precision error) comes not from the solution to another equation, but is just given as a numerical value: `values = {x: 1e16, y: 1, z: 1e16}`.

Reply:
My problem is that the example feels very contrived. Substituting the solution into the original equation is a great example of where you are very likely to get loss of precision issues, almost by definition. And it doesn't actually require you to have large values like 1e16 for that to occur. 

Really I would prefer if this wasn't necessary and it just worked correctly automatically with normal `subs`, but that requires some improvements to `subs`.

Reply:
Can you outline an example that wouldn't feel contrived? I think the point here is to alert visitors to the gotcha by demonstrating it and how to avoid it.

Certainly it would be best if this gotcha didn't exist, so this workaround wasn't necessary.

Reply:
I think substituting into the original equation isn't contrived, and it's also one of the more likely situations the gotcha would arise, since you're almost guaranteed to have loss of significance if the original expression has multiple terms in it.

Reply:
Substituting into the original equation doesn't seem to exhibit the gotcha--it gives identically zero, as you would expect:

```py
>>> from sympy import cos, nsolve, Symbol
>>> x = Symbol('x')
>>> f = cos(x) - x
>>> x_value = nsolve(f, x, 1)
>>> x_value
0.739085133215161
>>> f.subs(x, x_value)
0
```

Reply:
Maybe it's too subtle, but the point is that the expression *shouldn't* be identically zero. The value 0.739085133215161 is only an approximation to the true solution, which is infinitely many digits long, so substituting it into the expression should produce a value which is close to 0, but not exactly equal to it. The reason 0 is returned is because the naive substitution isn't smart enough to increase the precision of the operation to produce the actual value, which you can see from the `evalf(subs)` example above is around 1e-17. More specifically, in the default precision (15 digits), cos(0.739085133215161) just happens to produce exactly 0.739085133215161, so if you subtract them, you get 0. 

Another way to look at this is to say that the solution `0.739085133215161` is accurate to within 1e-16, since when plugged back into the original equation you get `-5.12757857962640e-17`. This is expected since the default precision in `nsolve` is 15, meaning the answer should be accurate to 15 digits (within 1e-15). Similarly, if you increase the precision requested, the result is accurate to within that precision:

```py
>>> x_value = nsolve(cos(x) - x, x, 1, prec=40)
>>> (cos(x) - x).evalf(subs={x: x_value})
-8.91241325272802e-42
```

(of course, it's not actually necessary for the user to check this, as `nsolve` does so automatically, but that could be seen as a motivating reason why you would do it)